### ISSUE TITLE
Parameter geom_type with value MULTILINESTRING does not work

### ISSUE THREAD

**Issue by PetrDlouhy** [makinacorpus/django-leaflet#183]
This is regression between versions `0.19.0` and `0.20.0`.  If I set in `LeafletWidget.attrs` `"geom_type": 'MULTILINESTRING'` to the options, the leaflet component loads only partially with following javascript error message:
```
Uncaught Unsupported geometry type: multiPolyline
_editionLayer	@	leaflet.forms.js:213
addTo	@	leaflet.forms.js:112
id_track_map_callback	@	(index):273
L.Map.djangoMap	@	leaflet.extras.js:234
loadmap	@	(index):286
```

Not all the controls are loaded in the map:
![snimek obrazovky_2017-03-21_08-13-37](https://cloud.githubusercontent.com/assets/156755/24136281/4f1281fe-0e0e-11e7-9b3c-48efcbcd697e.png)

**Comment by Gagaro** [comment]
MultiPolyline has been removed in leaflet 1.0.0 (https://github.com/Leaflet/Leaflet/blob/master/CHANGELOG.md#10-beta1-july-14-2015).

The fix should be done there: https://github.com/makinacorpus/django-leaflet/blob/master/leaflet/static/leaflet/leaflet.forms.js#L87-L88

**Comment by PetrDlouhy** [comment]
@Gagaro Thanks for putting this into context. I think, that the fix is done wrong. I found, that even `multiPolyline` is not in Leaflet anymore (although it is in documentation): http://stackoverflow.com/questions/31062687/l-multipolyline-does-not-show-anything-on-map-leaflet

**Comment by Gagaro** [comment]
I made a quick fix which seems to work, but I'm not sure this is the best way to handle it.

https://github.com/makinacorpus/django-leaflet/commit/36552a18ea3838eabcf6fef91aaa87726310d2af

@leplatrem , could you review that plz? It's a very small commit.

**Comment by Gagaro** [comment]
@PetrDlouhy You can test with this commit: https://github.com/makinacorpus/django-leaflet/commit/36552a18ea3838eabcf6fef91aaa87726310d2af

It should work, but map created with 0.19.0 considers all the lines in the MultiPolyline to be on the same layer, so trying to delete one polyline delete all the line. It works correctly when you add new line or on new map though.

**Comment by PetrDlouhy** [comment]
@Gagaro: Nice, the widget works now. But it produces invalid type of geometry. It doesn't process through `clean` method of `MultiLineStringField` - there is no value in `cleaned_data` and in errors is `{'track': ['Invalid geometry type.']} `.

With `0.19.0` it produced:
```
{
  "type": "MultiLineString",
  "coordinates": [
    [
      [
        14.407796859741,
        50.096247592266
      ],
      [
        14.41689491272,
        50.090741238315
      ]
    ]
  ]
}
```

And now it produces:
```
{
  "type": "GeometryCollection",
  "geometries": [
    {
      "type": "LineString",
      "coordinates": [
        [
          14.341278076172,
          50.092393210939
        ],
        [
          14.437408447266,
          50.087106698082
        ]
      ]
    }
  ]
}
```

**Comment by Gagaro** [comment]
Thanks for the feedback. I need to see what `multiPolygon` should produce as well then.

**Comment by Gagaro** [comment]
`MULTIPOLYGON` uses `MultiPolygon`, which doesn't exist anymore. I have an error when I try to replace it with `Polygon`.

**Comment by Gagaro** [comment]
MultiPolygon still seems to actually work..

**Comment by Gagaro** [comment]
Another fix, which should work and have the correct behaviour: https://github.com/makinacorpus/django-leaflet/commit/15356b2792b325e9836ce5eb3adf318d4cc4863f

@PetrDlouhy Could you test this?
@leplatrem Could you review?

Thanks :+1:

**Comment by PetrDlouhy** [comment]
@Gagaro Nice. This seems to work superbly. I didn't find any unexpected behaviour even if I switched between `django_leaflet` versions. Thank you very much.

**Comment by Gagaro** [comment]
@leplatrem Fast review possible for https://github.com/makinacorpus/django-leaflet/commit/15356b2792b325e9836ce5eb3adf318d4cc4863f?

Thanks :cat: .

**Comment by leplatrem** [comment]
To be honest I can't review this efficiently without trying it out. So
please go ahead if you have positive feedback​ from users ;)

**Comment by Gagaro** [comment]
merged in 15356b2792b325e9836ce5eb3adf318d4cc4863f
